Formula/openlitespeed.rb

```diff
@@ -11,19 +11,122 @@ class Openlitespeed < Formula
-    args << "--with-openssl=#{Formula["openssl"].opt_prefix}"
+    args << "--enable-http2=no" if build.without? "http2"
+    args << "--enable-redis=yes" if build.with? "redis"
+
+    if build.with? "zlib"
+      args << "--with-zlib=#{Formula["zlib"].opt_prefix}"
+    end
+
+    if build.with? "expat"
+      args << "--with-expat=#{Formula["expat"].opt_prefix}"
+      args << "--with-expat-inc=#{Formula["expat"].opt_prefix}/include"
+      args << "--with-expat-lib=#{Formula["expat"].opt_prefix}/lib"
+    end
+
     system "./configure", *args
+    system "make"
+
+    if build.with? "php7"
+      inreplace buildpath/"dist/admin/misc/build_admin_php.sh" do |s|
+        # PHP
+        s.gsub! "PHP_VERSION=5\.6\.18", "PHP_VERSION=7.0.7"
+
+        # LSAPI
+        s.gsub! "LSAPI_VERSION=6\.6", "LSAPI_VERSION=6.9"
+      end
+    end
+
     system "make", "install"
   end
+
+  def caveats; <<-EOS.undent
+    === PHP Compilation Note ===
+
+    As of this writing there is no safe and sustainable approach to instruct
+    openlitespeed's `make install` to NOT compile PHP, if you want to use
+    PHP-FPM or some other variant of PHP, you'd have to do it manually
+
+    === Start OLS manually ===
+
+    Use this:
+      LSWS_HOME=$(brew --prefix openlitespeed) lswsctrl start
+
+    `LshttpdMain::getServerRoot` requires either
+      LSWS_HOME or LSHTTPD_HOME
+
+    without either of them, you'll get this error:
+      "Can't determine the Home of LiteSpeed Web Server, exit!"
+
+    === / ===
+    EOS
+  end
+
+  plist_options :startup => false, :manual => "lswsctrl start"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>EnvironmentVariables</key>
+      <dict>
+        <key>LSWS_HOME</key>
+        <string>#{prefix}</string>
+      </dict>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/lswsctrl</string>
+        <string>start</string>
+      </array>
+      <key>RunAtLoad</key>
+      <true/>
+      <key>StandardErrorPath</key>
+      <string>#{prefix}/logs/launchd-error.log</string>
+      <key>StandardOutPath</key>
+      <string>#{prefix}/logs/launchd.log</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  def post_install
+    (prefix/"logs").mkpath
+    (prefix/"admin/cgid").mkpath
+
+    inreplace opt_bin/"lswsctrl.open", "RESTART_LOG=\"\$BASE_DIR\/\.\.\/logs\/lsrestart.log\"", "RESTART_LOG=\"#{prefix}\/logs\/lsrestart.log\""
+
+    inreplace prefix/"conf/httpd_config.conf", "lsphp5", "lsphp7"
+
+    mv "#{prefix}/fcgi-bin/lsphp5", "#{prefix}/fcgi-bin/lsphp7"
+  end
+
+  test do
+    assert File.exist?(opt_bin/"lswsctrl.open"), "Executable `lswsctrl` not found"
+    assert File.exist?(opt_bin/"openlitespeed"), "Executable `openlitespeed` not found"
```

Comment:
I could add:

` lswsctrl start -> status -> stop `  
although, the main entry point for adding vhosts is through the webadmin gui, so, `stop` part might be unnecessary :|